commit b6c8ce922a72e786a467c5233dde203d133de46e
Author: Benjamin Gilbert <bgilbert@backtick.net>
Date:   Sat Sep 23 00:57:14 2023 -0500

    meson: avoid linking with -pthread if we don't have pthreads
    
    Meson always returns -pthread in dependency('threads') on non-MSVC
    compilers.  Fix a link error when building on MinGW without winpthreads.

diff --git a/meson.build b/meson.build
index f822fb5476a3..5c896ac71b42 100644
--- a/meson.build
+++ b/meson.build
@@ -485,6 +485,9 @@ if dep_threads.found() and cc.has_header('pthread.h')
   else
     pthreads_found = true
   endif
+else
+  # Avoid linking with -pthread if we don't actually have pthreads
+  dep_threads = null_dep
 endif
 
 if pthreads_found
